src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs

```diff
@@ -200,7 +199,34 @@ public async Task<string> SubmitPledge(CreatePostRequest request)
 
             return _encodingService.Encode(pledgeId, EncodingType.PledgeId);
         }
+        public async Task RejectApplications(RejectApplicationsPostRequest request)
+        {
+            var serviceRequest = new SetRejectApplicationsRequest
+            {
+                UserId = _userService.GetUserId(),
+                UserDisplayName = _userService.GetUserDisplayName(),
+                ApplicationsToReject = GetDecodedApplicationsIds(request.ApplicationsToReject).ToList()
+            };
+
+            await _pledgeService.RejectApplications(serviceRequest, request.AccountId, request.PledgeId);
+        }
+
+        public async Task<RejectApplicationsViewModel> GetRejectApplicationsViewModel(RejectApplicationsRequest request)
+        {
+            var applicationsList = await _pledgeService.GetRejectApplications(request.AccountId, request.PledgeId);
 
+            if(!applicationsList.Applications.Any())
```

Comment:
I'm still not keen on this method.

Firstly, we have two `return new xxxViewModel` statements, which seems unnecessary. Could we not just have one? The only difference seems to be whether the `DasAccountNames` property is populated or not.

Secondly, some of the code has been moved into a method which is only used once, which doesn't seem worth it. The method in that code has nested foreach loops so isn't easily read (there's also a typo in a var name).

Reply:
I have updated that method now. removed private method and used your suggested code.